src/controllers/accounts.js

```diff
@@ -88,15 +95,18 @@ exports.putResource = function * putResource () {
   yield db.transaction(function * (transaction) {
     existed = yield Account.findByName(name, { transaction })
     if (existed) {
-      existed.setDataExternal(self.body)
+      existed.setDataExternal(account)
       yield existed.save({ transaction })
     } else {
-      yield Account.createExternal(self.body, { transaction })
+      yield Account.createExternal(account, { transaction })
     }
   })
 
   log.debug((existed ? 'updated' : 'created') + ' account name ' + name)
 
   this.body = this.body.getDataExternal()
+  if (!this.req.user.is_admin) {
+    delete this.body.is_disabled
```

Comment:
Don't really like the idea of having too many different versions of the schema. There is the user object, the public user object, the public user object without `is_disabled`...

My vote is not to delete the `is_disabled` property here. Makes it easier to write clients for the API and isn't hurting anybody.